autorel/builders/dist_tarball_builder.py

```diff
@@ -0,0 +1,74 @@
+"""
+    module : dist_tarball
+    This module is responsible for building syslog-ng from source
+    in a docker environment and providing functionality of exporting
+    the generated distribution tarball.
+"""
+import os
+import glob
+import tempfile
+from shutil import copyfile
+
+
+class DistTarballBuilder(object):
+    """
+        Responsible for downloading souce code and carry out
+        build process to generate the distribution tarball.
+    """
+    def __init__(self, input_directory, image_name, version):
+        self._input_directory = input_directory
+        self._image_name = image_name
+        self._version = version
+
+    @property
+    def image(self):
+        return self._image_name
+
+    @property
+    def input_directory(self):
+        """
+            This directory serves as the mount volume
+        """
+        return self._input_directory
+
+    @property
+    def build_commands(self):
+        """
+            Returns the list of commands to be executed to
+            build the distribution tarball
+        """
+        commands = [
+            "(cd /home/syslog-ng && pip install -r requirements.txt)",
```

Comment:
The `Release` class will clone the code in a temporary directory and the builder expects an argument of `input_directory` (https://github.com/black-perl/syslog-ng-autorel/blob/develop/autorel/builders/debian_source_builder.py#L39)  and this input_directory is mounted here https://github.com/black-perl/syslog-ng-autorel/blob/develop/autorel/utils/docker_cli.py#L107